SwiftCompilerSources/Sources/Optimizer/Utilities/Devirtualization.swift

```diff
@@ -33,7 +33,11 @@ private func devirtualize(destroy: some DevirtualizableDestroy, _ context: some
   if !type.isMoveOnly {
     return true
   }
-  precondition(type.isNominal, "non-copyable non-nominal types not supported, yet")
+
+  if !type.isNominal {
+    // E.g. a non-copyable generic function parameter
```

Comment:
This comment confuses me. Is a Copyable generic parameter “nominal”?

Reply:
I mean, e.g. `func foo<T: ~Copyable>(_ t: T)`. T is not copyable and not a nominal type